src/amo/components/AddonRecommendations/index.js

```diff
@@ -0,0 +1,106 @@
+/* @flow */
+import invariant from 'invariant';
+import * as React from 'react';
+import { Experiment, Variant } from 'react-ab';
+import cookie from 'react-cookie';
+import { compose } from 'redux';
+
+import tracking from 'core/tracking';
+import translate from 'core/i18n/translate';
+import type { AddonType } from 'core/types/addons';
+import type { I18nType } from 'core/types/i18n';
+
+export const TAAR_IMPRESSION_CATEGORY = 'AMO Addon / Recommendations Shown';
+export const TAAR_COHORT_COOKIE_NAME = 'TAARCohort';
+export const TAAR_COHORT_INCLUDED: 'TAAR_COHORT_INCLUDED'
+  = 'TAAR_COHORT_INCLUDED';
+export const TAAR_COHORT_EXCLUDED: 'TAAR_COHORT_EXCLUDED'
+  = 'TAAR_COHORT_EXCLUDED';
+
+export type cohortName = typeof TAAR_COHORT_INCLUDED | typeof TAAR_COHORT_EXCLUDED;
+
+type Props = {|
+  _cookie: typeof cookie,
+  _randomizer: Function,
+  _tracking: typeof tracking,
+  addon: AddonType | null,
+  i18n: I18nType,
+|};
+
+
+export class AddonRecommendationsBase extends React.Component<Props> {
+  static defaultProps = {
+    _cookie: cookie,
+    _randomizer: Math.random,
+    _tracking: tracking,
+  }
+
+  constructor(props: Props) {
+    super(props);
+    // This will prevent the component from rendering on the server, by only
+    // setting this to true in componentDidMount, which is only run on the
+    // client.
+    this.shouldRender = false;
```

Comment:
Why exactly do we want to use client side rendering only? Because `react-ab` does not really support SSR or is there something else I am missing?

Reply:
The problem is that the call to `tracking.sendEvent` in `onChoice` fails if it's called on the server, and we don't end up recording the impression. We thought this was a good approach, but have since discussed it and decided it would be better to adjust the tracking module to be smart enough to not try to do some thing it cannot do. I was initially worried that we might miss recording the impression if `render` is only called on the server, but I now gather that it will always be called on the client after the file is downloaded.